clients/drcachesim/simulator/cache_simulator.h

```diff
@@ -56,19 +64,23 @@ class cache_simulator_t : public simulator_t
     // Create a cache_t object with a specific replacement policy.
     virtual cache_t *create_cache(std::string policy);
 
-    // Currently we only support a simple 2-level hierarchy.
-    // XXX i#1715: add support for arbitrary cache layouts.
-
     cache_simulator_knobs_t knobs;
 
     // Implement a set of ICaches and DCaches with pointer arrays.
     // This is useful for implementing polymorphism correctly.
-    cache_t **icaches;
-    cache_t **dcaches;
+    cache_t **l1_icaches;
+    cache_t **l1_dcaches;
+
+    // Vector of LLCs.
+    std::vector<cache_t*> llcaches;
+
+    // Hash map mapping a cache's name to a pointer to it.
+    // All caches in the hierarchy are included in the map.
+    std::map<string, cache_t*> all_caches;
```

Comment:
Please comment on the ordering needed here: i.e., why it's not std::unordered_map

Reply:
Changed to unordered map.